docs/user-guide/ingest-data/for-observerbility/loki.md

```diff
@@ -0,0 +1,103 @@
+  * `X-Greptime-Log-Table-Name`: `<table_name>` (optional) - The table name to store the logs. If not provided, the default table name is `loki_logs`.
+
+### Example Code
+
+```hcl
+loki.source.file "greptime" {
+  targets = [
+    {__path__ = "/tmp/foo.txt"},
+  ]
+  forward_to = [loki.write.greptime_loki.receiver]
+}
+
+loki.write "greptime_loki" {
+    endpoint {
+        url = "${GREPTIME_SCHEME:=http}://${GREPTIME_HOST:=greptimedb}:${GREPTIME_PORT:=4000}/v1/loki/api/v1/push"
+        headers  = {
+          "x-greptime-db-name" = "${GREPTIME_DB:=public}",
+          "x-greptime-log-table-name" = "${GREPTIME_LOG_TABLE_NAME:=loki_demo_logs}",
+        }
+    }
+    external_labels = {
+        "job" = "greptime"
+        "from" = "alloy"
+    }
+}
+```
+
+You can run the following command to check the data in the table:
+
+```sql
+mysql> select * from loki_demo_logs;
++----------------------------+------------------------+--------------+-------+----------+
+| greptime_timestamp         | line                   | filename     | from  | job      |
++----------------------------+------------------------+--------------+-------+----------+
+| 2024-11-25 11:02:31.256251 | Greptime is very cool! | /tmp/foo.txt | alloy | greptime |
++----------------------------+------------------------+--------------+-------+----------+
+1 row in set (0.01 sec)
+
+mysql> desc loki_demo_logs;
++--------------------+---------------------+------+------+---------+---------------+
+| Column             | Type                | Key  | Null | Default | Semantic Type |
++--------------------+---------------------+------+------+---------+---------------+
+| greptime_timestamp | TimestampNanosecond | PRI  | NO   |         | TIMESTAMP     |
+| line               | String              |      | YES  |         | FIELD         |
+| filename           | String              | PRI  | YES  |         | TAG           |
+| from               | String              | PRI  | YES  |         | TAG           |
+| job                | String              | PRI  | YES  |         | TAG           |
++--------------------+---------------------+------+------+---------+---------------+
+5 rows in set (0.00 sec)
+
+mysql> show create table loki_demo_logs\G
+*************************** 1. row ***************************
+       Table: loki_demo_logs
+Create Table: CREATE TABLE IF NOT EXISTS `loki_demo_logs` (
+  `greptime_timestamp` TIMESTAMP(9) NOT NULL,
+  `line` STRING NULL,
+  `filename` STRING NULL,
+  `from` STRING NULL,
+  `job` STRING NULL,
+  TIME INDEX (`greptime_timestamp`),
+  PRIMARY KEY (`filename`, `from`, `job`)
+)
+
+ENGINE=mito
+WITH(
+  append_mode = 'true'
+)
+1 row in set (0.00 sec)
+```
+
+## Data Model
+
+The Loki logs data model is mapped to the GreptimeDB data model according to the following rules:
```

Comment:
please explain more about the rules:
- how to specify tags and time index.
- the data type conversion rules